### PR TITLE
Show comments on code example

### PR DESCRIPTION
**Pull request by Imperatorn**
I'm trying to think like a complete newbie. 
I think we often forget that some people actually come to D as a first language.

1. tee = Perform a transparent operation
image

2. walkLength = optimized version of length
3. byLine = Get an input range set up to read one line at a time

### CODE REVIEW

File: index.dd [hunk 1]
```diff
@@ -43,8 +43,13 @@ void main()
...
 
     auto sum = 0.0;
-    auto count = stdin.byLine
-        .tee!(l => sum += l.length).walkLength;
+    auto count = stdin
+        //Get an input range set up to read one line at a time
+        .byLine
+        //Perform a transparent operation 
+        .tee!(l => sum += l.length)
+        //Optimized version of length
```
Comment (thewilsonator) [inline index.dd:51]: this comment is misleading. In this case `.length` is not available since it isn't known in advance.

File: index.dd [hunk 2]
```diff
@@ -43,8 +43,13 @@ void main()
     import std.range, std.stdio;
 
     auto sum = 0.0;
-    auto count = stdin.byLine
-        .tee!(l => sum += l.length).walkLength;
+    auto count = stdin
+        //Get an input range set up to read one line at a time
+        .byLine
+        //Perform a transparent operation
```
Comment (thewilsonator) [inline index.dd:49]: you might want to add something like "as in the unix command `tee`"

**Review by thewilsonator — COMMENTED** [review]
`

File: index.dd (hunk 1 above)
Comment (Imperatorn) [reply index.dd:51]: How would you describe walkLength instead?

**Comment by thewilsonator** [comment]
```
Searching for trailing whitespace
./index.dd:52:        .walkLength;
```

File: index.dd (hunk 1 above)
Comment (thewilsonator) [reply index.dd:51]: It iterates the range and counts for each iteration. I'm not sure how to best word that.

Comment (Imperatorn) [reply index.dd:51]: What about 
"Best-effort implementation of length" or "Best-effort implementation of length for ranges"?

Or maybe remove the comment? 🤔

Comment (thewilsonator) [reply index.dd:51]: I think it would be fine with no comment, `walkLength` is a lot easier to figure out from the context than `tee`.

**Review by thewilsonator — APPROVED** [review]

**Pull request merged**